sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -530,6 +530,26 @@ def test_ordinal_encoder(X):
     assert_array_equal(enc.fit_transform(X), exp)
 
 
+
+@pytest.mark.parametrize("X", [
+    [['abc', 2, 55], ['def', 1, 55], [np.nan, np.nan, np.nan]],
+    np.array([[10, 2, 55], [20, 1, 55], [np.nan, np.nan, np.nan]]),
+    np.array([['a', 'B', 'cat'],
+              ['b', 'A', 'cat'],
+              [np.nan, np.nan, np.nan]], dtype=object)
+    ], ids=['mixed', 'numeric', 'object'])
+def test_ordinal_encoder_handle_missing(X):
+    enc = OrdinalEncoder(encode_missing=True)
+    exp = np.array([[0, 1, 0],
+                    [1, 0, 0],
+                    [2, 2, 1]], dtype='float64')
+    assert_array_equal(enc.fit_transform(X), exp)
+    enc = OrdinalEncoder(encode_missing=False)
+    exp = np.array([[0, 1, 0],
+                    [1, 0, 0],
+                    [np.nan, np.nan, np.nan]], dtype='float64')
+    assert_array_equal(enc.fit_transform(X), exp)
+
```

Comment:
Can you also add a case where you have different data for fit and transform? Eg the case where there are no missing values during fitting, but only during transform (this will then need to interact with `handle_unknown` I think ?)